python/ray/_raylet.pyx

```diff
@@ -806,9 +807,19 @@ cdef void execute_task(
                             deserialize_args, function_descriptor,
                             name_of_concurrency_group_to_execute)
                     else:
-                        args = (ray._private.worker.global_worker
-                                .deserialize_objects(
-                                    metadata_pairs, object_refs))
+                        # Defer task cancellation (SIGINT) until after the task argument
+                        # deserialization context has been left.
+                        # NOTE (Clark): We defer SIGINT until after task argument
+                        # deserialization completes to keep from interrupting
+                        # non-reentrant imports that may be re-entered during error
+                        # serialization or storage.
+                        # See https://github.com/ray-project/ray/issues/30453.
+                        # NOTE (Clark): Signal handlers can only be registered on the
+                        # main thread.
+                        with DeferSigint.create_if_main_thread():
```

Comment:
should hijack this in ray core, or only in the arrow serialization plugin?

Reply:
This is a generic problem for Ray Core, so I don't think that we'd want to only fix it for the arrow serialization plugin.